### ISSUE TITLE
Creates Empty SGF and then analyzes :-)

### ISSUE THREAD

**Issue by laefsky** [lightvector/leela-analysis#12]
Thank you for what I know was a lot of work and which I hope will improve my go playing and analysis
I know some python, the python installed as a system default is Anaconda 3.6 so I am using a winpython 2.7
to run the program mybat.bat is my running script.  When I run the program seems to write a zero length
sgf file and then proceed to do its analysis.  Can you offer any advice.  Is there an email address or phone number where I can contact you directly. My email is laefsky AT comcast DOT net

Windows Bat File Follows

.\python.exe E:\leela-analysis-master\sgfanalyze.py -v3 E:\downloads\testa.sgf --leela E:\Leela0110GTP\Leela0110GTP_OpenCL.exe > test2.sgf
 my email is : laefsky AT comcast DOT net

**Comment by lightvector** [comment]
Does the ">" operation work differently in windows? The idea is supposed to be that you redirect the output of the program to a file, because the output to stdout is the SGF contents. So that's not working if it creates the file and then still outputs the SGF to console.

Or maybe you didn't wait long enough for the analysis to finish? If what you're seeing on the console is information about the progress of the analysis, then of course the SGF file will be empty for a while, it hasn't finished analyzing it yet to output anything. You need to just wait until it's done and when it's done the SGF will be nonempty.

**Comment by laefsky** [comment]
Thank You for the rapid reply.  As far as I know > redirects STDOUT to a file and >> redirects STDOUT and appends.  Some differences are that I need to run python on the file from an executable as the system Python is Anaconda 3.6.  Is it supposed to create a null foo.sgf immediately and then append the results of the analysis? Are there flags to place
The intermediate analysis files in a given directory?   Is the fact that I’m running python from a command shell and only then redirecting output mean that I am not capturing the results of the python program output.
How long should the analysis run with a multicore i7 and GTX 1080 GPU?  I am creating the analysis intermediate files and watching program output with -v3 in a python shell window but I don’t know whether I am effectively redirecting output from the python subprocess?

Many thanks for your excellent program and for your anticipated help.

--Ira Laefsky

**Comment by lightvector** [comment]
Yes, it's supposed to create an empty file immediately, this has nothing to do with any of my code, that's what ">" does, it creates a file and then over time as the program outputs, it puts that output in the file.

Sounds like you should just wait for the analysis to finish? It will output all the sgf analysis when the whole program is done running, which then should get redirected to your file.

Edit: It will show a (very approximate) time estimate, if you watch it. Normally it should take at least some hours, more if you told it to use more time per move.

**Comment by laefsky** [comment]
Hi:

 

But the file is not gradually increasing in size; or does that only happen when the program halts?

 

Thank you

--Ira

**Comment by lightvector** [comment]
It only happens when the program halts.

**Comment by laefsky** [comment]
Hi & Thanks:  On the verbose output -v3 the number of visited positions has three values which appear to go up and down (that is the number of visited positions for listing 1, 2, 3 do not increase monotonically and I am running

For over one hour on a fairly simple ~220 odd move game on fairly robust hardware that I use for running Keras and Tensorflow an i7 5820K CPU at 3.3 Ghz and a Nvidia GTX 1080 GPU?   Should I let it continue to run?

 

Thanks

--Ira

**Comment by laefsky** [comment]
Hi:

 

The python window and process has halted but the sgf file is still of zero length. What can I try now or how can I get further diagnostics?

 

Thank you

--Ira

 

From: Ira Laefsky [mailto:laefsky@comcast.net] 
Sent: Tuesday, March 6, 2018 9:37 AM
To: 'lightvector/leela-analysis' <reply@reply.github.com>; 'lightvector/leela-analysis' <leela-analysis@noreply.github.com>; 'Ira Laefsky' <laefsky@comcast.net>
Cc: 'Author' <author@noreply.github.com>
Subject: RE: [lightvector/leela-analysis] Creates Empty SGF and then analyzes :-) (#12) 

 

Hi & Thanks:  On the verbose output -v3 the number of visited positions has three values which appear to go up and down (that is the number of visited positions for listing 1, 2, 3 do not increase monotonically and I am running

For over one hour on a fairly simple ~220 odd move game on fairly robust hardware that I use for running Keras and Tensorflow an i7 5820K CPU at 3.3 Ghz and a Nvidia GTX 1080 GPU?   Should I let it continue to run?

 

Thanks

--Ira

**Comment by lightvector** [comment]
Try running it without redirecting to any file at all. You should see some
text at the very end that looks like the contents of an SGF file. That's
the analyzed SGF produced by the script. (If you open an existing sgf file
you have in a text editor you can see what sgf format looks like). Or if
not, then presumably you should see some error it produced for why the
analysis crashed.

In the case where you do see the SGF-like output at the very end, it's your
job to get that into an actual SGF file, whether by manually copying and
pasting or figuring out how output redirecting works on Windows. I don't
know how output redirecting actually works on Windows so if the analysis
script is behaving as it should and the only problem is getting the output
to go to the right place, I can't really help you there. If the script
itself is failing and crashing, then I can help if you paste the error
backtrace here.

Also, while debugging, try running with --help to see the arguments and
provide the arguments to make it only run on a limited move range of your
SGF and/or make it spend less time per move, so you can test faster.

**Comment by laefsky** [comment]
Thank You I Will Try Your Suggestions and Get Back To You

--Ira

**Comment by laefsky** [comment]
--I wasn't getting (in Windows 10) any sgf-like text at the end of python2 window. I decided to try installing in my Ubuntu 16.04 Partition and immediately got the help file for leala-analyze and it looks like so far I am getting proper redirection although the final sgf hasnt yet been created with my linux instantiation. I suspect one or both was going wrong with my windows 10 example: 1. the bat file was not appropriately directing inputs and outputs and file names. 2. the fact that I needed a sub-process to fire off python 2.7 and the "pipes" were not connected properly. I would be interested in consulting with someone who is both familiar with go leela and leela-analyze as well as the command shell at batch language in windows. Please keep all of the previous items in the thread available for anyone who wishes to try leela-analyze with windows.

--Thank you
Ira

**Comment by laefsky** [comment]
I have loaded leela-analysis and leela11 into my linux partition.  It works almost out-of-the-box in Linux.  Apparently one or both problems in Windows 10 are: improped batch .bat script (but not just redirect as the end of the anlysis in the python shell does not result in a pdf. 2. Because my default python in windows is Anaconda 3.6 I cant make sgfanalyze a standalone executable and when I attempt to run python 2.7 (Winpython) from the batch

Command in the Command Shell t he subprocess doesn’t do the right things with the files and redirection.  I would appreciate help from someone who knows these programs and Windows 10.  Please keep the discussion comments open for further debugging.

 

Thank you

--Ira

 

From: Ira Laefsky [mailto:laefsky@comcast.net] 
Sent: Tuesday, March 6, 2018 7:57 PM
To: 'lightvector/leela-analysis' <reply@reply.github.com>; 'lightvector/leela-analysis' <leela-analysis@noreply.github.com>; 'Ira Laefsky' <laefsky@comcast.net>
Cc: 'Author' <author@noreply.github.com>
Subject: RE: [lightvector/leela-analysis] Creates Empty SGF and then analyzes :-) (#12)

 

Thank You I Will Try Your Suggestions and Get Back To You

--Ira
